src/status_im/hardwallet/core.cljs

```diff
@@ -49,17 +93,151 @@
   [{:keys [db]} enabled?]
   {:db (assoc-in db [:hardwallet :nfc-enabled?] enabled?)})
 
-(fx/defn navigate-to-connect-screen [{:keys [db] :as cofx}]
+(fx/defn status-hardwallet-option-pressed [{:keys [db] :as cofx}]
   (fx/merge cofx
             {:hardwallet/check-nfc-enabled    nil
              :hardwallet/register-card-events nil
-             :db                              (assoc-in db [:hardwallet :setup-step] :begin)}
+             :db                              (-> db
+                                                  (assoc-in [:hardwallet :setup-step] :begin)
+                                                  (assoc-in [:hardwallet :on-card-connected] nil)
+                                                  (assoc-in [:hardwallet :pin :on-verified] nil))}
             (navigation/navigate-to-cofx :hardwallet-connect nil)))
 
 (fx/defn success-button-pressed [cofx]
   ;; login not implemented yet
 )
 
+(fx/defn change-pin-pressed
+  [{:keys [db] :as cofx}]
+  (let [card-connected? (get-in db [:hardwallet :card-connected?])
+        pin-retry-counter (get-in db [:hardwallet :application-info :pin-retry-counter])
+        enter-step (if (zero? pin-retry-counter) :puk :current)]
```

Comment:
Duplicated logic?

Reply:
@jeluard moved to separate fn

Reply:
@jeluard yes, there is some duplication. But it's just 2 simple operation which are not worth moving into separate fn, unless you can come up with some meaningful fn name for that.